debian/ubuntu-advantage-tools.preinst

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+set -e
+
+remove_old_config_fields() {
+    PREVIOUS_PKG_VER="$1"
+    if dpkg --compare-versions "$PREVIOUS_PKG_VER" le "27.8"; then
+        if grep -q "^license_check_log_file:" /etc/ubuntu-advantage/uaclient.conf; then
+            sed -i '/^license_check_log_file:.*$/d' /etc/ubuntu-advantage/uaclient.conf || true
+        fi
+    fi
+}
+
+case "$1" in
+    upgrade)
```

Comment:
Using preinst is nice, however there is a corner case: the user uninstalled u-a-t <= 27.8 *without purging*, and now wants to install it. Not purging means that the conf files of the package are preserved.

In this case the preinst script is called with as:
```
preinst install old-version
```
which is not handled at the moment. See https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html#summary-of-ways-maintainer-scripts-are-called for doc on this.

Basically we need to also handle the `install)` case, e.g. like here:

https://sources.debian.org/src/unattended-upgrades/2.8/debian/preinst/

Reply:
@paride Thanks for noticing this! And that example was really helpful. Just pushed an update to resolve this